Add serviceAccountName input/search dropdown in deployment workload

jordojordo opened the pull request:
rancher/dashboard#1814 
Created form component to allow searching of service account names within the selected namespace. Currently, new service names must be created manually. If a user types in a service name that doesn't exist they will receive a message warning them.
Tested locally with multiple namespaces and service names.

nwmac commented on components/form/ServiceNameSelect.vue:
```diff
@@ -0,0 +1,284 @@
+    },
+
+    clearSearch(event) {
+      if (event.pointerId === 1) {
```
I don't think pointerId is right here - I think we can remove this if

nwmac reviewed (changes requested):
@jordojordo I updated your PR and changed the trash icon to a 'Clear' button. Also fixed an issue where the page would reload when you hit the clear button.

In addition to the comment, there is one other issue:

- The value fo serviceAccountName is not populated on edit. You can see this by editing a workload, setting a service account name and then editing the config of the workload - the service account name is empty instead of showing the correct name

nwmac commented:
@jordojordo Can you update this PR?

richard-cox commented on components/form/ServiceNameSelect.vue:
```diff
@@ -0,0 +1,256 @@
+          />
+        </div>
+      </div>
+      <button class="btn btn-sm clear-btn role-secondary col span-1" @click="clearSearch($event);">
```
This shows up on the 'view config' page (so needs to take `mode` into account, and be hidden if is `_VIEW`)

richard-cox commented on components/form/LabeledSearchSelect.vue:
```diff
@@ -0,0 +1,378 @@
+<script>
```
There's a super duper amount of C&P from LabeledSelect in LabeledSearchSelect. The former has a `searchable` property, can that be used? If not can the functionality in the later be merged in to the former?

richard-cox commented on components/form/ServiceNameSelect.vue:
```diff
@@ -0,0 +1,256 @@
+        {{ t('generic.clear') }}
+      </button>
+    </div>
+    <template v-if="serviceNameNew">
```
The initial view shows the `create it manually` warning. When there's no service supplied it shouldn't show
![image](https://user-images.githubusercontent.com/18697775/142904550-9a904adf-b128-4956-8115-31af733be71d.png)

jordojordo replied:
That was a simple issue where `serviceNameNew` was searching for an empty string.

richard-cox commented on components/form/ServiceNameSelect.vue:
```diff
@@ -0,0 +1,256 @@
+  height: 30px;
+}
+
+.input-container {
```
This has all come from `InputWithSelect`, is it needed?

richard-cox commented on components/form/ServiceNameSelect.vue:
```diff
@@ -0,0 +1,256 @@
+  }
+}
+
+.servicename-select ::v-deep {
```
Is there a way to avoid using `::v-deep`?

richard-cox commented on components/form/ServiceNameSelect.vue:
```diff
@@ -0,0 +1,256 @@
+    },
+
+    change() {
+      const selectedOption = this.options.find(option => option.metadata.id === this.selected.serviceAccount);
```
This throws an error whenever the user manually enters text

```
client.js?06a0:97 TypeError: Cannot read properties of null (reading 'serviceAccount')
    at eval (ServiceNameSelect.vue?0417:86)
```

richard-cox commented on components/form/ServiceNameSelect.vue:
```diff
@@ -0,0 +1,256 @@
+        <div
+          :class="{ 'select-after': !selectBeforeText }"
+          class="servicename-select input-container container-flex"
+          @input="change"
```
Is this needed when there's the `input` handler on `LabeledSearchSelect`?

richard-cox commented on components/form/ServiceNameSelect.vue:
```diff
@@ -0,0 +1,256 @@
+          @input="change"
+        >
+          <LabeledSearchSelect
+            v-if="selectLabel"
```
This if is a bit odd, is it needed here?

richard-cox commented on components/form/ServiceNameSelect.vue:
```diff
@@ -0,0 +1,256 @@
+    <div class="row">
+      <div class="col span-5">
+        <div
+          :class="{ 'select-after': !selectBeforeText }"
```
Is this needed?

richard-cox commented on components/form/ServiceNameSelect.vue:
```diff
@@ -0,0 +1,256 @@
+      default: false,
+    },
+
+    searchable: {
```
The plumbing for this isn't passed through to `LabeledSearchSelect`

jordojordo commented:
@richard-cox I have thoroughly cleaned this PR. Got rid of the LabeledSearchSelect component as it was completely unnecessary, I reused the LabeledSelect component to achieve the same results. Fixed the minor bugs that were all around, and removed the unused css.

richard-cox reviewed (changes requested):
Two minor tweaks and one bug .. the 'Clear' feature doesn't get applied when editing a deployment with an existing value
